sympy/core/basic.py

```diff
@@ -105,6 +105,7 @@ class Basic(Printable, metaclass=ManagedProperties):
     is_Point = False
     is_MatAdd = False
     is_MatMul = False
+    is_Set = False
```

Comment:
We already have too many of these `is_` attributes. It would be better not to add more. We can just do `isinstance(obj, Set)`.

Reply:
But earlier you said, we shouldn't import Set in core.

Reply:
Oh, yes. I see what the problem is now.

The special casing of Set should be implemented by overriding `is_eq` as described here:
https://github.com/sympy/sympy/blob/25252ee4457461f7dfe568d1f495b4df6f7ca93d/sympy/core/relational.py#L1260

The code to override this can be in the sets module because it uses multiple dispatch.

Reply:
Done!